src/iop/atrous.c

```diff
@@ -572,33 +599,28 @@ static void eaw_synthesize(float *const out, const float *const in, const float
 
 #if defined(__SSE2__)
 static void eaw_synthesize_sse2(float *const out, const float *const in, const float *const detail,
-                                const float *thrsf, const float *boostf, const int32_t width,
-                                const int32_t height)
+                                const float *thrsf, const float *boostf, const size_t width,
+                                const size_t height)
 {
-  const __m128 threshold = _mm_set_ps(thrsf[3], thrsf[2], thrsf[1], thrsf[0]);
-  const __m128 boost = _mm_set_ps(boostf[3], boostf[2], boostf[1], boostf[0]);
+  const __m128 threshold = _mm_loadu_ps(thrsf);
```

Comment:
This is loading threshold not in the same order as original. Likewise for boost.

Reply:
yes it does :-) the `_mm_set_ps` takes its arguments in reverse order, as the `_mm_load_ps`, the natural order can be used with `_mm_setr_ps. See https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_set_ps&expand=4895

Reply:
My bad, sorry :)